libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/config/SyncsConfig.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2017-present, salesforce.com, inc.
+ * All rights reserved.
+ * Redistribution and use of this software in source and binary forms, with or
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * - Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.mobilesync.config
+
+import android.content.Context
+import com.salesforce.androidsdk.mobilesync.manager.SyncManager
+import com.salesforce.androidsdk.mobilesync.target.SyncDownTarget
+import com.salesforce.androidsdk.mobilesync.target.SyncUpTarget
+import com.salesforce.androidsdk.mobilesync.util.MobileSyncLogger
+import com.salesforce.androidsdk.mobilesync.util.SyncOptions
+import com.salesforce.androidsdk.mobilesync.util.SyncState
+import com.salesforce.androidsdk.smartstore.config.StoreConfig
+import com.salesforce.androidsdk.smartstore.store.SmartStore
+import com.salesforce.androidsdk.util.JSONObjectHelper
+import com.salesforce.androidsdk.util.ResourceReaderHelper
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Class encapsulating syncs definition.
+ *
+ * Config expected in a resource or assets file in JSON with the following:
+ * {
```

Comment:
Optional: I was just curious but Kdoc supports ```preformatted code``` using backticks just like markdown does.

Reply:
Right now we only run / publish javadoc back to [github.io](https://forcedotcom.github.io/SalesforceMobileSDK-Android/index.html). But we should think about using kdoc and updating inlined documentation in the code accordingly. Maybe a new story / epic ?